File: src/main/java/ast/ASTArrayFuncType.java
package ast;

import java.util.List;

public class ASTArrayFuncType extends ASTFuncType {
    int length;

    public ASTArrayFuncType(ASTFileLocation loc, List<ASTVarDecl> args, ASTType ret_type, int length){
        super(loc, args, ret_type);
        this.length = length;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTArrayFuncType: (ret_type, args[])");
        ret_type.print(i+1);
        for(var p = 0; p < args.size(); p++){
            args.get(p).print(i+1);
        }
    }
}

File: src/main/kotlin/ast/ASTArrayFuncType.kt
package ast

class ASTArrayFuncType(loc: ASTFileLocation, args: List<ASTVarDecl>, ret_type: ASTType?, var length: Int) : ASTFuncType(loc, args, ret_type) {

    override fun print(i: Int) {
        printIndent(i)
        System.out.println("ASTArrayFuncType: (ret_type, args[])")
        ret_type!!.print(i + 1)
        for (p in 0 until args.size) {
            args[p].print(i + 1)
        }
    }
}